main2.py

```diff
@@ -76,11 +76,11 @@ def process_kinesis(kinesis, file_list_path):
     flp = open(file_list_path, 'r')
     for line in flp:
         file_path = line.strip('\n').strip('\r\n')
-        fp = open(file_path, 'r')
-        for content in fp:
-            if content.find("kinesis") != -1:
-                record = load_kinesis_record(content)
-                save_kinesis_record(kinesis, record)
+        with open(file_path, 'r') as fp:
+            for content in fp:
+                if content.find("kinesis") != -1:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you do not need the position returned by the [`find()`](https://docs.python.org/3/library/stdtypes.html#str.find) function. Consider using the operators [`in` and `not in`](https://docs.python.org/3/reference/expressions.html#membership-test-details) to check for simple membership, or more specific operations [`startswith()`](https://docs.python.org/3/library/stdtypes.html#str.startswith) and [`endswith()`](https://docs.python.org/3/library/stdtypes.html#str.endswith) to check for prefix and suffix relationships with the string.